From 4f453f20f2deb3127a48d3c218bb0750571cd48b Mon Sep 17 00:00:00 2001
From: "Liu, Kai1" <kai1.liu@intel.com>
Date: Tue, 28 Jun 2022 13:17:58 +0800
Subject: [FEATURE] Dequeue the next back buffer in
 droid_swap_buffers

This can reduce latency of GLSurfaceView, which has render
thread to draw frame.

Signed-off-by: Liu, Kai1 <kai1.liu@intel.com>
Change-Id: Iaf5077b7f60959c997e0cbcbdf4678c0fe7426d9
Signed-off-by: Marc Mao <marc.mao@intel.com>
---
 src/egl/drivers/dri2/platform_android.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/egl/drivers/dri2/platform_android.c b/src/egl/drivers/dri2/platform_android.c
index 245850a5b86..cc401ad3f1f 100644
--- a/src/egl/drivers/dri2/platform_android.c
+++ b/src/egl/drivers/dri2/platform_android.c
@@ -1066,6 +1066,13 @@ droid_swap_buffers(_EGLDisplay *disp, _EGLSurface *draw)
       draw->ActiveRenderBuffer = draw->RequestedRenderBuffer;
    }
 
+   /* try to dequeue the next back buffer */
+   if (!dri2_surf->buffer && !droid_window_dequeue_buffer(dri2_surf)) {
+      _eglLog(_EGL_WARNING, "Could not dequeue buffer from native window");
+      dri2_surf->base.Lost = EGL_TRUE;
+      return -1;
+   }
+
    return EGL_TRUE;
 }
 
-- 
2.33.1

